include/alpaka/mem/view/Accessor.hpp

```diff
@@ -0,0 +1,222 @@
+/* Copyright 2021 Bernhard Manfred Gruber
+
+ * Permission to use, copy, modify, and/or distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED “AS IS” AND ISC DISCLAIMS ALL WARRANTIES WITH
+ * REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL ISC BE LIABLE FOR ANY
+ * SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR
+ * IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#pragma once
+
+#include <alpaka/dim/Traits.hpp>
+#include <alpaka/mem/view/Traits.hpp>
+
+namespace alpaka
+{
+    namespace internal
+    {
+        template<typename T>
+        auto asBytePtr(T* p)
+        {
+            return reinterpret_cast<char*>(p);
+        }
+        template<typename T>
+        auto asBytePtr(const T* p)
+        {
+            return reinterpret_cast<const char*>(p);
+        }
+    } // namespace internal
+
+    enum class AccessMode
+    {
+        read_only,
+        read_write
+    };
+
+    template<typename Pointer, typename Value, typename Idx, typename Dim>
+    struct Accessor;
```

Comment:
I think the Dim type should become an integral type. That should make accessor declarations leaner:
```c++
template<typename Pointer, typename Value, typename Idx, int Dim>
struct Accessor;

...

alpaka::Accessor<bool*, bool, alpaka::Idx<TAcc>, alpaka::DimInt<1>>
vs.
alpaka::Accessor<bool*, bool, alpaka::Idx<TAcc>, 1>

```

Reply:
If I remember correctly it is in alpaka a type to avoid compiler conversion errors/warnings. If it is possible to switch to an integral type only I would suggest using `std::uint32_t`, `int` is variable in the number of bits and a dimension should be positive only.

Reply:
I chose a `std::size_t` now.